packages/paste-website/src/components/shortcodes/live-preview/index.tsx

```diff
@@ -1,64 +1,113 @@
+
+  const handleToggleCodeEditor = (): void => {
+    setViewCode(!viewCode);
+  };
+
   return (
     <Box marginBottom="space110" data-cy="live-preview">
-      <LiveProvider
-        code={children}
-        scope={scope}
-        language={language}
-        theme={CodeblockTheme}
-        disabled={disabled}
-        noInline={noInline}
-      >
+      <LiveProvider code={children} scope={scope} language={language} theme={CodeblockTheme} noInline={false} disabled>
         <Box
-          paddingLeft="space50"
-          paddingRight="space50"
-          paddingTop="space80"
-          paddingBottom="space80"
+          padding="space70"
           borderColor="colorBorderWeak"
           borderStyle="solid"
           borderWidth="borderWidth20"
-          borderBottomWidth="borderWidth0"
           backgroundColor="colorBackgroundBody"
           borderTopLeftRadius="borderRadius20"
           borderTopRightRadius="borderRadius20"
+          position="relative"
+          overflowX="auto"
         >
           <ReactLivePreview />
         </Box>
         <Box
-          css={{padding: '2px 10px', backgroundColor: '#011627'}}
+          backgroundColor="colorBackgroundBodyInverse"
+          paddingY="space110"
+          paddingX="space70"
           borderBottomLeftRadius="borderRadius20"
           borderBottomRightRadius="borderRadius20"
           borderColor="colorBorderWeak"
           borderStyle="solid"
           borderWidth="borderWidth20"
           borderTopWidth="borderWidth0"
           fontSize="fontSize20"
+          position="relative"
+          height={isSmallCodeVariant || viewCode ? 'auto' : '112px'}
+          overflow="hidden"
         >
-          <LiveEditor style={{fontFamily: pasteTheme.fonts.fontFamilyCode}} />
+          {/* corrects an inbuilt 10px margin style so we can accurately use token values */}
+          <LiveEditor
+            style={{
+              margin: '-10px',
+              fontFamily: pasteTheme.fonts.fontFamilyCode,
+              backgroundColor: pasteTheme.backgroundColors.colorBackgroundBodyInverse,
+            }}
+          />
+          {isSmallCodeVariant || viewCode ? null : <CodeBlockOverlayShadow />}
+          {isSmallCodeVariant ? null : (
+            <Box position="absolute" top="space40" right="space40">
+              <Button variant="secondary" size="small" onClick={handleToggleCodeEditor}>
+                {viewCode ? (
+                  <>
+                    <HideIcon decorative />
+                    <span>Hide code</span>
+                  </>
+                ) : (
+                  <>
+                    <ShowIcon decorative />
+                    <span>View code</span>
+                  </>
+                )}
+              </Button>
+            </Box>
+          )}
+          {isSmallCodeVariant || viewCode ? (
+            <Box position="absolute" bottom="space40" right="space40">
```

Comment:
To make sure screen reader users know that the text of the button changed, this copy/copied button should match the one on icons list page: https://github.com/twilio-labs/paste/blob/1b4111d4a47ea3af35a26605d8174d4290a04f4e/packages/paste-website/src/components/icons-list/IconCard.tsx#L53

maybe it should be its own component...

Reply:
Done.